src/components/tag/Tag.tsx

```diff
@@ -1,58 +1,116 @@
 import React from "react";
-import styled from "styled-components";
+import styled, { css } from "styled-components";
 import { SVGIcon } from "../svgicon";
 import theme from "../../theme";
+import { darken } from "polished";
+import { space } from "styled-system";
 
-interface TagProps {
+type TagProps = {
   color?: string;
+  colorText?: string;
   text?: string;
+  icon?: string;
   onClick?: () => void;
   showClose?: boolean;
-}
+};
 
-export const StyledTag = styled.div`
-  display: inline-flex;
+export const StyledTag = styled.div<TagProps>`
+  display: flex;
+  float: left;
   align-items: center;
-  text-transform: capitalize;
-  text-align: center;
-  font-family: ${theme.fonts.body};
-  font-size: ${theme.fontSizes.sm};
-  line-height: ${theme.lineHeight};
   padding: 0 ${theme.spacing.spacing03};
-  margin: ${theme.spacing.spacing01};
-  border-radius: 1rem;
-  color: ${theme.colors.lights[0]};
-  background-color: ${({ color }) =>
-    color ? theme.colors[color] || color : theme.colors.lights[3]};
-  cursor: ${({ onClick }) => onClick && "pointer"};
-  span {
-    width: 0.8rem;
-    height: 0.8rem;
-    display: block;
-    font-size: 0.5rem;
-    font-weight: bold;
-    margin-left: 0.4rem;
-    margin-bottom: 0.2rem;
-    transform: translate(2px, 1px);
+  font-size: ${theme.fontSizes.sm};
+  line-height: 1.5rem;
+  background: ${theme.colors.lights[2]};
+  border-radius: ${theme.radius.xxl};
+  margin: 0 4px 4px 0;
+
+  .tag__icon,
+  .tag__close-button {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: 1rem;
+    height: 1rem;
+  }
+
+  .tag__icon {
+    margin: 0 0.25rem 0 -0.375rem;
+  }
+
+  .tag__close-button {
+    border-radius: ${theme.radius.xxl};
+    background: ${darken(0.1, theme.colors.lights[2])};
+    margin: 0 -0.375rem 0 0.5rem;
+
+    svg {
+      width: 0.625rem;
+      height: 0.625rem;
+    }
   }
+
   &:hover {
-    opacity: ${({ onClick }) => onClick && "0.8"};
+    background: ${darken(0.05, theme.colors.lights[2])};
+    cursor: pointer;
+    transition: all 0.2s ease-in-out;
```

Comment:
transitions should be inside constant, you can extract "0.2s ease-in-out"

Reply:
transition: ${theme.transition};